catalog/core/catalog-core-solr/src/test/java/ddf/catalog/source/solr/provider/SolrProviderExtensibleMetacards.java

```diff
@@ -0,0 +1,323 @@
+    return customMetacard;
+  }
+
+  private Metacard createDescriptionMetacard() {
+    descriptionDescriptors.add(
+        new AttributeDescriptorImpl(Metacard.ID, true, true, true, false, BasicTypes.STRING_TYPE));
+    descriptionDescriptors.add(
+        new AttributeDescriptorImpl(
+            DESCRIPTION_FIELD, true, true, true, false, BasicTypes.STRING_TYPE));
+    MetacardTypeImpl mType = new MetacardTypeImpl("custom1", descriptionDescriptors);
+    MetacardImpl customMetacard = new MetacardImpl(mType);
+    customMetacard.setAttribute(DESCRIPTION_FIELD, DESCRIPTION_VALUE);
+    return customMetacard;
+  }
+
+  private Metacard createTypesMetacard() {
+
+    typesDescriptors.add(
+        new AttributeDescriptorImpl(Metacard.ID, true, true, true, false, BasicTypes.STRING_TYPE));
+    typesDescriptors.add(
+        new AttributeDescriptorImpl(doubleField, true, true, false, false, BasicTypes.DOUBLE_TYPE));
+    typesDescriptors.add(
+        new AttributeDescriptorImpl(floatField, true, true, false, false, BasicTypes.FLOAT_TYPE));
+    typesDescriptors.add(
+        new AttributeDescriptorImpl(intField, true, true, false, false, BasicTypes.INTEGER_TYPE));
+    typesDescriptors.add(
+        new AttributeDescriptorImpl(longField, true, true, false, false, BasicTypes.LONG_TYPE));
+    typesDescriptors.add(
+        new AttributeDescriptorImpl(byteField, false, true, false, false, BasicTypes.BINARY_TYPE));
+    typesDescriptors.add(
+        new AttributeDescriptorImpl(
+            booleanField, true, true, false, false, BasicTypes.BOOLEAN_TYPE));
+    typesDescriptors.add(
+        new AttributeDescriptorImpl(dateField, true, true, false, false, BasicTypes.DATE_TYPE));
+    typesDescriptors.add(
+        new AttributeDescriptorImpl(geoField, true, true, false, false, BasicTypes.GEO_TYPE));
+    typesDescriptors.add(
+        new AttributeDescriptorImpl(shortField, true, true, false, false, BasicTypes.SHORT_TYPE));
+    typesDescriptors.add(
+        new AttributeDescriptorImpl(
+            objectField, false, true, false, false, BasicTypes.OBJECT_TYPE));
+
+    MetacardTypeImpl mType = new MetacardTypeImpl("numbersMT", typesDescriptors);
+
+    MetacardImpl customMetacard = new MetacardImpl(mType);
+    customMetacard.setAttribute(doubleField, doubleFieldValue);
+    customMetacard.setAttribute(floatField, floatFieldValue);
+    customMetacard.setAttribute(intField, intFieldValue);
+    customMetacard.setAttribute(longField, longFieldValue);
+    customMetacard.setAttribute(byteField, byteFieldValue);
+    customMetacard.setAttribute(booleanField, true);
+    customMetacard.setAttribute(dateField, dateFieldValue);
+    customMetacard.setAttribute(geoField, geoFieldValue);
+    customMetacard.setAttribute(shortField, shortFieldValue);
+    customMetacard.setAttribute(objectField, objectFieldValue);
+
+    return customMetacard;
+  }
+
+  @Test
+  public void queryById() throws UnsupportedQueryException {
```

Comment:
❓ Should we change the name to indicate that we're testing wildcard on ID, not a specific ID?